lib/configurators/aws/aws.go

```diff
@@ -327,58 +333,51 @@ func (c *ConfiguratorConfig) CheckAndSetDefaults() error {
 	if !c.Flags.Manual {
 		var err error
 
-		if c.AWSSession == nil {
-			c.AWSSession, err = awssession.NewSessionWithOptions(awssession.Options{
-				SharedConfigState: awssession.SharedConfigEnable,
-				Config: aws.Config{
-					UseFIPSEndpoint: useFIPSEndpoint,
-				},
-			})
-			if err != nil {
-				return trace.Wrap(err)
-			}
+		awsConfig, err := config.LoadDefaultConfig(ctx,
+			config.WithUseFIPSEndpoint(useFIPSEndpoint),
+		)
+		if err != nil {
+			return trace.Wrap(err)
 		}
 
-		if c.AWSSTSClient == nil {
-			c.AWSSTSClient = sts.New(c.AWSSession)
+		if c.stsClient == nil {
+			c.stsClient = sts.NewFromConfig(awsConfig)
 		}
-		if c.AWSIAMClient == nil {
-			c.AWSIAMClient = iam.New(c.AWSSession)
+		if c.iamClient == nil {
+			c.iamClient = iam.NewFromConfig(awsConfig)
 		}
 		if c.Identity == nil {
-			c.Identity, err = awslib.GetIdentityWithClient(context.Background(), c.AWSSTSClient)
+			c.Identity, err = awslib.GetIdentityWithClientV2(context.Background(), c.stsClient)
 			if err != nil {
 				return trace.Wrap(err)
 			}
 		}
-		if c.AWSSSMClients == nil {
-			c.AWSSSMClients = make(map[string]ssmiface.SSMAPI)
+
+		if c.ssmClients == nil {
+			c.ssmClients = make(map[string]ssmClient)
 			for _, matcher := range c.ServiceConfig.Discovery.AWSMatchers {
 				if !slices.Contains(matcher.Types, types.AWSMatcherEC2) {
 					continue
 				}
 				for _, region := range matcher.Regions {
-					if _, ok := c.AWSSSMClients[region]; ok {
+					region := region
```

Comment:
Why is this necessary?

Reply:
region is captured by the closure below (sdk v2 config uses the closure, v1 did not).
With go >=1.22 it works without reassignment, but on older go versions I think it would be bugged. branch/v14 is still on 1.21. I have no plans to backport this to v14 but I figured it didn't hurt to guard against it

Reply:
But `o.Region = region` is a string assignment, which makes a copy.

Are you sure this is necessary?

Reply:
evaluation is delayed until the config is actually used, but actually I see that we're applying the option within the loop, so you're right. I'll remove that reassignment then.